teeio-validator/library/cxl_ide_test_lib/test_case/ide_km_common.c

```diff
@@ -0,0 +1,209 @@
+{
+  TEEIO_PRINT(("\n"));
+  TEEIO_PRINT(("Print host cxl ide status\n"));
+  cxl_dump_ide_status(upper_port->cxl_data.memcache.cap_headers, upper_port->cxl_data.memcache.cap_headers_cnt, upper_port->cxl_data.memcache.mapped_memcache_reg_block);
+
+  TEEIO_PRINT(("\n"));
+  TEEIO_PRINT(("Print device cxl ide status.\n"));
+  cxl_dump_ide_status(lower_port->cxl_data.memcache.cap_headers, lower_port->cxl_data.memcache.cap_headers_cnt, lower_port->cxl_data.memcache.mapped_memcache_reg_block);
+}
+
+// setup cxl ide stream
+bool cxl_setup_ide_stream(void *doe_context, void *spdm_context,
+                          uint32_t *session_id, uint8_t *kcbar_addr,
+                          uint8_t stream_id,
+                          uint8_t port_index,
+                          ide_common_test_port_context_t *upper_port,
+                          ide_common_test_port_context_t *lower_port,
+                          bool skip_ksetgo, uint32_t config_bitmap)
+{
+  bool result;
+  uint8_t kp_ack_status;
+  libspdm_return_t status;
+  INTEL_KEYP_CXL_ROOT_COMPLEX_KCBAR *kcbar_ptr;
+  cxl_ide_km_aes_256_gcm_key_buffer_t rx_key_buffer = {0};
+  cxl_ide_km_aes_256_gcm_key_buffer_t tx_key_buffer = {0};
+  INTEL_KEYP_KEY_SLOT keys = {0};
+
+  kcbar_ptr = (INTEL_KEYP_CXL_ROOT_COMPLEX_KCBAR *)kcbar_addr;
+
+  // ide_km_key_prog in RX
+  result = libspdm_get_random_number(sizeof(rx_key_buffer.key), (void *)rx_key_buffer.key);
+  if (!result)
+  {
+    TEEIO_DEBUG((TEEIO_DEBUG_ERROR, "libspdm_get_random_number for rx_key_buffer failed.\n"));
+    return false;
+  }
+
+  status = cxl_ide_km_key_prog(
+      doe_context, spdm_context,
+      session_id, stream_id,
+      CXL_IDE_KM_KEY_DIRECTION_RX | CXL_IDE_KM_KEY_IV_DEFAULT | CXL_IDE_KM_KEY_SUB_STREAM_CXL,
+      port_index, // port_index
+      &rx_key_buffer, &kp_ack_status);
+
+  if (LIBSPDM_STATUS_IS_ERROR(status))
+  {
+    TEEIO_DEBUG((TEEIO_DEBUG_ERROR, "cxl_ide_km_key_prog RX failed with status=0x%08x\n", status));
+    TEEIO_ASSERT(false);
+    return false;
+  }
+  TEEIO_DEBUG((TEEIO_DEBUG_INFO, "key_prog RX - %02x\n", kp_ack_status));
+
+  // ide_km_key_prog in TX
+  result = libspdm_get_random_number(sizeof(tx_key_buffer.key), (void *)tx_key_buffer.key);
+  if (!result)
+  {
+    TEEIO_DEBUG((TEEIO_DEBUG_ERROR, "libspdm_get_random_number for tx_key_buffer failed.\n"));
+    return false;
+  }
+
+  status = cxl_ide_km_key_prog(
+      doe_context, spdm_context,
+      session_id, stream_id,
+      CXL_IDE_KM_KEY_DIRECTION_TX | CXL_IDE_KM_KEY_IV_DEFAULT | CXL_IDE_KM_KEY_SUB_STREAM_CXL,
+      port_index,
+      &tx_key_buffer, &kp_ack_status);
+  if (LIBSPDM_STATUS_IS_ERROR(status))
+  {
+    TEEIO_DEBUG((TEEIO_DEBUG_ERROR, "cxl_ide_km_key_prog TX failed with status=0x%08x\n", status));
+    TEEIO_ASSERT(false);
+    return false;
+  }
+  TEEIO_DEBUG((TEEIO_DEBUG_INFO, "key_prog TX - %02x\n", kp_ack_status));
+
+  // TODO
```

Comment:
what is this TODO?

Reply:
This TODO shall be deleted, together with other un-used lines in the codes.